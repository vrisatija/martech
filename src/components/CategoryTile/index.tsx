import React, { FC } from 'react';
import { useNavigate } from 'react-router-dom';
import Images from '../../assets';

interface IProps {
  categoryImg: string;
  title: JSX.Element;
  redirectionRoute: string;
}

const CategoryTile: FC<IProps> = ({ categoryImg, title, redirectionRoute }) => {
  const navigate = useNavigate();
  return (
    <button
      className="rounded-3xl w-76.25 h-91.75 transform transition duration-500 hover:scale-105"
      style={{
        backgroundImage: `url("${Images.cardBackground}")`,
      }}
      data-testid=""
      type="button"
      onClick={() => navigate(redirectionRoute)}
    >
      <div className="px-10 pt-10">
        <img src={categoryImg} alt="search artwork" />
        <div className="flex flex-col items-center mt-4 text-white font-poppins-semi-bold text-center">
          {title}
        </div>
      </div>
    </button>
  );
};

export default CategoryTile;
