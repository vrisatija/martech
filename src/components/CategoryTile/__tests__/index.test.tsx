import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import {
  render, fireEvent, screen, waitFor,
} from '@testing-library/react';
import CategoryTile from '..';

const mockedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedNavigate,
}));

describe('CategoryTile Component', () => {
  it('renders correctly', () => {
    const CategoryTileFragment = render(
      <BrowserRouter>
        <CategoryTile
          categoryImg=""
          title={<>test</>}
          redirectionRoute="/test"
        />
      </BrowserRouter>,
    ).asFragment();
    expect(CategoryTileFragment).toMatchSnapshot();
  });

  it('redirects user on button click', async () => {
    render(
      <BrowserRouter>
        <CategoryTile
          categoryImg=""
          title={<>test</>}
          redirectionRoute="/test"
        />
      </BrowserRouter>,
    ).asFragment();
    await fireEvent.click(screen.getByRole('button'));
    await waitFor(() => expect(mockedNavigate).toHaveBeenCalledWith('/test'));
  });
});
