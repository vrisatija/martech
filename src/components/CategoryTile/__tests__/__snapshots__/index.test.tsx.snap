// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryTile Component renders correctly 1`] = `
<DocumentFragment>
  <button
    class="rounded-3xl w-76.25 h-91.75 transform transition duration-500 hover:scale-105"
    data-testid=""
    style="background-image: url(cardBackground.svg);"
    type="button"
  >
    <div
      class="px-10 pt-10"
    >
      <img
        alt="search artwork"
        src=""
      />
      <div
        class="flex flex-col items-center mt-4 text-white font-poppins-semi-bold text-center"
      >
        test
      </div>
    </div>
  </button>
</DocumentFragment>
`;
